File: app/src/main/java/com/polestar/charging/ui/cluster/view/renderer/MarkerCache.java
package com.polestar.charging.ui.cluster.view.renderer;


import com.amap.api.maps.model.Marker;

import java.util.HashMap;
import java.util.Map;

/**
 * A cache of markers representing individual ClusterItems.
 */
public class MarkerCache<T> {
    private Map<T, Marker> mCache = new HashMap<>();
    private Map<Marker, T> mCacheReverse = new HashMap<>();

    public Marker get(T item) {
        return mCache.get(item);
    }

    public T get(Marker m) {
        return mCacheReverse.get(m);
    }

    public void put(T item, Marker m) {
        mCache.put(item, m);
        mCacheReverse.put(m, item);
    }

    public void remove(Marker m) {
        T item = mCacheReverse.get(m);
        mCacheReverse.remove(m);
        mCache.remove(item);
    }
}

File: app/src/main/java/com/polestar/charging/ui/cluster/MarkerCache.kt
package com.polestar.charging.ui.cluster

import com.amap.api.maps.model.Marker
import com.polestar.charging.ui.cluster.base.ClusterItem
import java.util.HashMap

class MarkerCache<T> {
    private val mCache = HashMap<T, Marker>()
    private val mCacheReverse = HashMap<Marker, T>()

    fun get(item: T): Marker? {
        return mCache.get(item)
    }


/*    private Map<T, Marker> mCache = new HashMap<>();
    private Map<Marker, T> mCacheReverse = new HashMap<>();

    public Marker get(T item) {
        return mCache.get(item);
    }*/



    fun get(item: Marker) = mCacheReverse.get(item)

    fun put(item: T, marker: Marker) {
        mCache[item] = marker
        mCacheReverse[marker] = item
    }

    fun remove(marker: Marker) {
        val t = mCacheReverse.get(marker)
        mCacheReverse.remove(marker)
        mCache.remove(t)
    }
}